### PR TITLE
Sensor2

### PR DESCRIPTION
**Pull request by asadate**
Add thermal and magnetic sensor specifications and updated power.json to address some of the changes recently introduced on the unit object definition (No more typ/min/max in property names since the values are captured in the unit object).

### CODE REVIEW

File: part-spec/power.json [hunk 1]
```diff
@@ -10,11 +10,9 @@
...
         "componentID",
         "regulatorTopology",
-        "vinMin",
-        "vinMax",
-        "voutMin",
-        "voutMax",
-        "maxLoadCurrent"
+        "vin",
+        "vout",
+        "LoadCurrent"
```
Comment (rsnancollas) [inline part-spec/power.json]: loadCurrent

Keep case consistent

File: part-spec/power.json [hunk 2]
```diff
@@ -31,141 +29,54 @@
...
+          "description": "output voltage variation from minimum input voltage to maximum input voltage ",
           "$ref": "definitions.json#/unit"
         },
         "currentConsumption": {
           "description": "current consumption of a device",
-          "type": "array",
-          "items": {
-            "$ref": "definitions.json#/currentConsumption"
-          }
+          "$ref": "definitions.json#/currentConsumption"
```
Comment (rsnancollas) [inline part-spec/power.json:70]: "comment": "units of amps"

File: part-spec/power.json [hunk 3]
```diff
@@ -31,141 +29,54 @@
...
-          "$ref": "definitions.json#/unit"
-        },
-        "lineRegulationMax": {
-          "description": "maximum output voltage variation from minimum input voltage to maximum input voltage ",
-          "$ref": "definitions.json#/unit"
-        },
-        "lineRegulationMin": {
-          "description": "minimum output voltage variation from minimum input voltage to maximum input voltage ",
+        "lineRegulation": {
+          "description": "output voltage variation from minimum input voltage to maximum input voltage ",
```
Comment (rsnancollas) [inline part-spec/power.json:63]: "comment": "units of volts"

File: part-spec/power.json [hunk 4]
```diff
@@ -31,141 +29,54 @@
...
-          "$ref": "definitions.json#/unit"
-        },
-        "voutAccuracyTyp": {
-          "description": "typical output voltage variation at no load",
-          "$ref": "definitions.json#/unit"
-        },
-        "voutAccuracyMax": {
-          "description": "maximum output voltage variation at no load",
+        "loadRegulation": {
+          "description": "output voltage variation from no load to full load ",
```
Comment (rsnancollas) [inline part-spec/power.json:58]: "comment": "units of volts"

File: part-spec/power.json [hunk 5]
```diff
@@ -31,141 +29,54 @@
...
+        "loadCurrent": {
+          "description": "load current supported by a device",
           "comment": "units of amps",
           "$ref": "definitions.json#/unit"
         },
-        "loadCurrentMax": {
-          "description": "maximum load current supported by a device",
-          "comment": "units of amps",
+        "voutAccuracy": {
+          "description": "output voltage variation at no load",
```
Comment (rsnancollas) [inline part-spec/power.json:53]: "comment": "units of volts"

File: part-spec/power.json [hunk 6]
```diff
@@ -206,165 +114,64 @@
...
-          "$ref": "definitions.json#/unit"
-        },
-        "currentLimitTyp": {
-          "description": "typical high sustained output current threshold beyond which the output of a device starts drooping",
-          "comment": "units of amps",
-          "$ref": "definitions.json#/unit"
-        },
-        "currentLimitMin": {
-          "description": "minimum high sustained output current threshold beyond which the output of a device starts drooping",
+        "LoadCurrent": {
```
Comment (rsnancollas) [inline part-spec/power.json]: loadCurrent

keep case consistent

File: part-spec/power.json [hunk 7]
```diff
@@ -206,165 +114,64 @@
...
+          "description": "load current supported by a device",
           "comment": "units of amps",
           "$ref": "definitions.json#/unit"
         },
-        "currentLimitMax": {
-          "description": "maximum high sustained output current threshold beyond which the output of a device starts drooping",
+        "currentLimit": {
+          "description": "high sustained output current threshold beyond which the output of a device starts drooping",
```
Comment (rsnancollas) [inline part-spec/power.json]: delete "high"

File: part-spec/power.json [hunk 8]
```diff
@@ -206,165 +114,64 @@
...
-          "$ref": "definitions.json#/unit"
-        },
-        "voutAccuracyMin": {
-          "description": "minimum output voltage variation at no load of a device",
-          "$ref": "definitions.json#/unit"
-        },
-        "voutAccuracyMax": {
-          "description": "maximum output voltage variation at no load of a device",
+        "voutAccuracy": {
+          "description": "output voltage variation at no load of a device",
```
Comment (rsnancollas) [inline part-spec/power.json:161]: "comment": "units of volts",

File: part-spec/power.json [hunk 9]
```diff
@@ -206,165 +114,64 @@
...
           "$ref": "definitions.json#/unit"
         },
-        "loadRegulationTyp": {
-          "description": "typical output voltage variation,from no load to full load, of a device ",
+        "loadRegulation": {
+          "description": "output voltage variation,from no load to full load, of a device ",
```
Comment (rsnancollas) [inline part-spec/power.json:166]: "comment": "units of volts",

File: part-spec/power.json [hunk 10]
```diff
@@ -206,165 +114,64 @@
...
           "$ref": "definitions.json#/unit"
         },
-        "loadRegulationMin": {
-          "description": "minimum output voltage variation,from no load to full load, of a device ",
+        "lineRegulation": {
+          "description": "output voltage variation,from minimum input voltage to maximum input voltage, of a device ",
```
Comment (rsnancollas) [inline part-spec/power.json:171]: "comment": "units of volts",

File: part-spec/power.json [hunk 11]
```diff
@@ -412,10 +213,9 @@
       "type": "object",
       "required": [
         "componentID",
-        "vinMin",
-        "vinMax",
-        "OutputCurrentMax",
-        "onResistanceTyp"
+        "vin",
+        "OutputCurrent",
```
Comment (rsnancollas) [inline part-spec/power.json]: outputCurrent

keep case consistent

File: part-spec/power.json [hunk 12]
```diff
@@ -439,48 +239,23 @@
...
-        "vinMax": {
-          "description": "maximum input voltage under which the part can be expected to reliabily operate ",
+        "vin": {
+          "description": "input voltage under which a device can be expected to reliabily operate",
           "comment": "units of volts",
           "$ref": "definitions.json#/unit"
         },
-        "OutputCurrentMax": {
-          "description": "maximum continuous DC cuurent supported by a device",
+        "OutputCurrent": {
```
Comment (rsnancollas) [inline part-spec/power.json]: outputCurrent

keep case consistent

File: part-spec/power.json [hunk 13]
```diff
@@ -439,48 +239,23 @@
...
+          "description": "continuous DC cuurent supported by a device",
           "comment": "units of amps",
           "$ref": "definitions.json#/unit"
         },
-        "oneResistanceTyp": {
-          "description": "typical FET on state resistance",
+        "oneResistance": {
```
Comment (rsnancollas) [inline part-spec/power.json]: onResistance

(not "**one**Resistance")

File: part-spec/power.json [hunk 14]
```diff
@@ -650,34 +379,18 @@
...
-          }
+          "$ref": "definitions.json#/currentConsumption"
         },
         "componentProtectionThresholds": {
           "description": "Thermal and power supply protection thresholds of a device",
-          "type": "array",
-          "items": {
-            "$ref": "icProtection.json#/componentProtectionThresholds"
-          }
+          "$ref": "icProtection.json#/componentProtectionThresholds"
```
Comment (rsnancollas) [inline part-spec/power.json]: definitions.json?

File: part-spec/power.json [hunk 15]
```diff
@@ -504,79 +279,33 @@
...
```
Comment (rsnancollas) [inline part-spec/power.json]: definitions.json?

File: part-spec/power.json [hunk 16]
```diff
@@ -697,96 +410,42 @@
...
-          "$ref": "definitions.json#/unit"
-        },
-        "vinMax": {
-          "description": "maximum input voltage under which a device can be expected to operate properly",
-          "comment": "units of volts",
-          "$ref": "definitions.json#/unit"
-        },
-        "voutTyp": {
-          "description": "typical output voltage a device can regulate",
+          "description": "input voltage under which a device can be expected to operate properly",
```
Comment (rsnancollas) [inline part-spec/power.json:429]: "vin", not "vinTyp", right?

File: part-spec/power.json [hunk 17]
```diff
@@ -951,206 +579,80 @@
...
-          "description": "typical charging current regulation accuracy of a device",
           "comment": "units of %",
           "$ref": "definitions.json#/unit"
         },
-        "batteryChargeCurrentAccuracyMin": {
-          "description": "minimum charging current regulation accuracy of a device",
-          "comment": "units of %",
+        "batteryChargeCurrent": {
+          "description": "charging current of a device",
+          "comment": "units of volts",
```
Comment (rsnancollas) [inline part-spec/power.json]: units of amps?

File: part-spec/power.json [hunk 18]
```diff
@@ -951,206 +579,80 @@
...
-          }
+          "$ref": "definitions.json#/currentConsumption"
         },
         "componentProtectionThresholds": {
           "description": "Thermal and power supply protection thresholds of a device",
-          "type": "array",
-          "items": {
-            "$ref": "icProtection.json#/componentProtectionThresholds"
-          }
+          "$ref": "icProtection.json#/componentProtectionThresholds"
```
Comment (rsnancollas) [inline part-spec/power.json]: definitions.json?

File: part-spec/power.json [hunk 19]
```diff
@@ -1176,18 +678,18 @@
           "description": "describes integrated fet current limits and rdson properties",
           "$ref": "#/$defs/powerFetProperities"
         },
-        "maxGateCapacitance": {
-          "description": "describes maximum gate capacitance supported on external fets",
+        "GateCapacitance": {
```
Comment (rsnancollas) [inline part-spec/power.json]: gateCapacitance

keep case consistent

File: part-spec/power.json [hunk 20]
```diff
@@ -1176,18 +678,18 @@
...
+          "description": "describes gate capacitance supported on external fets",
           "comment": "units of farads",
           "$ref": "definitions.json#/unit"
         },
-        "typInputSenseResistor": {
-          "description": "describes typical intput sense resistor value",
+        "InputSenseResistor": {
```
Comment (rsnancollas) [inline part-spec/power.json]: inputSenseResistor
keep case consistent

File: part-spec/power.json [hunk 21]
```diff
@@ -1176,18 +678,18 @@
...
+          "description": "describes intput sense resistor value",
           "comment": "units of ohms",
           "$ref": "definitions.json#/unit"
         },
-        "typBatterySenseResistor": {
-          "description": "describes typical battery sense resistor value",
+        "BatterySenseResistor": {
```
Comment (rsnancollas) [inline part-spec/power.json]: batterySenseResistor

keep case consistent

File: part-spec/sensor.json [hunk 22]
```diff
@@ -223,6 +223,175 @@
...
+            "type": "object",
+            "required": [
+                "componentID"
+            ],
+            "properties": {
+                "componentID": {
+                    "description": "common component identifying information, such as mpn",
+                    "$ref": "definitions.json#/componentID"
+                },
+                "type": {
```
Comment (rsnancollas) [inline part-spec/sensor.json]: maybe need a more descriptive name like

magneticSensingTechnology

File: part-spec/sensor.json [hunk 23]
```diff
@@ -223,6 +223,175 @@
...
+                    "description": "method by which magnetism is detected",
+                    "enum": [
+                        "gmr",
+                        "hall effect"
+                    ],
+                    "type": "string"
+                },
+                "outputType": {
+                    "description": "measurement output type",
+                    "enum": [
```
Comment (rsnancollas) [inline part-spec/sensor.json:248]: examples, not enum - are there any other options?

File: part-spec/sensor.json [hunk 24]
```diff
@@ -223,6 +223,175 @@
...
+                    "description": "magnetic flux density range over which the output voltage is linear ",
+                    "comment": "units of mT",
+                    "$ref": "definitions.json#/value"
+                },
+                "sensitivity": {
+                    "description": "this is the gain - amount of change in the output voltage for a change in the magnetic flux density",
+                    "comment": "units of mV/mT",
+                    "$ref": "definitions.json#/value"
+                },
+                "operatePoint": {
```
Comment (rsnancollas) [inline part-spec/sensor.json:274]: maybe call this fluxThreshold, unless operatingPoint is a standard term

File: part-spec/sensor.json [hunk 25]
```diff
@@ -223,6 +223,175 @@
...
+                    "description": "magnetic flux density threshold which causes the sensor output to turn on",
+                    "comment": "units of mT. Depending on the sensor's design, the operation point can cause the output to trip High or Low.",
+                    "$ref": "definitions.json#/value"
+                },
+                "releasePoint": {
+                    "description": "magnetic flux density threshold which causes the sensor output to turn off",
+                    "comment": "units of mT. Depending on the sensor's design, the release point can cause the output to trip High or Low.",
+                    "$ref": "definitions.json#/value"
+                },
+                "operatePointPolarity":{
```
Comment (rsnancollas) [inline part-spec/sensor.json]: maybe outputPolarity

"indicates whether the sensor output is active high or active low"

File: part-spec/sensor.json [hunk 26]
```diff
@@ -223,6 +223,175 @@
...
+            "type": "object",
+            "required": [
+                "componentID"
+            ],
+            "properties": {
+                "componentID": {
+                    "description": "common component identifying information, such as mpn",
+                    "$ref": "definitions.json#/componentID"
+                },
+                "type": {
```
Comment (rsnancollas) [inline part-spec/sensor.json]: maybe

"sensingTechnology"

You could also use this above, I would just choose something more descriptive than "type"

File: part-spec/sensor.json [hunk 27]
```diff
@@ -223,6 +223,175 @@
...
+                    "description": "method by which temperature is detected",
+                    "examples": [
+                        "rtd",
+                        "thermistor",
+                        "silicon bandgap"
+                    ],
+                    "type": "string"
+                },
+                "outputType": {
```
Comment (rsnancollas) [inline part-spec/sensor.json:339]: I'm not sure you need this and interface

File: part-spec/sensor.json [hunk 28]
```diff
@@ -223,6 +223,175 @@
...
+                    "description": "measurement output type",
+                    "enum": [
+                        "analog",
+                        "digital"
+                    ],
+                    "type": "string"
+                },
+                "interface": {
+                    "description": "interface(s) for communication to host",
+                    "enum": [
```
Comment (rsnancollas) [inline part-spec/sensor.json]: enum vs example

are there any other ways to communicate this?

File: part-spec/sensor.json [hunk 29]
```diff
@@ -223,6 +223,175 @@
...
+                        "analog"
+                    ],
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    }
+                },
+                "accuracy": {
+                    "description": "accuracy of temperature sensor",
+                    "comment": "units of degreesC",
```
Comment (rsnancollas) [inline part-spec/sensor.json]: degrees C

File: part-spec/sensor.json [hunk 30]
```diff
@@ -223,6 +223,175 @@
...
+                    "$ref": "definitions.json#/unit"
+                },
+                "temperatureRange": {
+                    "description": "range of temperature sensor",
+                    "comment": "units of degreesC",
```
Comment (rsnancollas) [inline part-spec/sensor.json:366]: degrees C

File: part-spec/sensor.json (hunk 22 above)
Comment (asadate) [reply part-spec/sensor.json]: done

File: part-spec/sensor.json (hunk 23 above)
Comment (asadate) [reply part-spec/sensor.json:248]: done

File: part-spec/sensor.json (hunk 24 above)
Comment (asadate) [reply part-spec/sensor.json:274]: This is a standard term

File: part-spec/sensor.json (hunk 25 above)
Comment (asadate) [reply part-spec/sensor.json]: done

File: part-spec/sensor.json (hunk 27 above)
Comment (asadate) [reply part-spec/sensor.json:339]: Some of these sensors still have different interfaces to the the host. As for the output, yes, some are digital vs analog.

File: part-spec/sensor.json (hunk 28 above)
Comment (asadate) [reply part-spec/sensor.json]: Good point. Especially with I3C becoming popular.

File: part-spec/sensor.json (hunk 29 above)
Comment (asadate) [reply part-spec/sensor.json]: done

File: part-spec/power.json (hunk 1 above)
Comment (asadate) [reply part-spec/power.json]: done

File: part-spec/power.json (hunk 5 above)
Comment (asadate) [reply part-spec/power.json:53]: added

File: part-spec/power.json (hunk 4 above)
Comment (asadate) [reply part-spec/power.json:58]: added

File: part-spec/power.json (hunk 3 above)
Comment (asadate) [reply part-spec/power.json:63]: added

File: part-spec/power.json (hunk 2 above)
Comment (asadate) [reply part-spec/power.json:70]: added

File: part-spec/power.json (hunk 6 above)
Comment (asadate) [reply part-spec/power.json]: changed

File: part-spec/power.json (hunk 7 above)
Comment (asadate) [reply part-spec/power.json]: done

File: part-spec/power.json (hunk 8 above)
Comment (asadate) [reply part-spec/power.json:161]: done

File: part-spec/power.json (hunk 9 above)
Comment (asadate) [reply part-spec/power.json:166]: added

File: part-spec/power.json (hunk 10 above)
Comment (asadate) [reply part-spec/power.json:171]: added

File: part-spec/power.json (hunk 11 above)
Comment (asadate) [reply part-spec/power.json]: done

File: part-spec/power.json (hunk 12 above)
Comment (asadate) [reply part-spec/power.json]: done

File: part-spec/power.json (hunk 13 above)
Comment (asadate) [reply part-spec/power.json]: Good catch. done

File: part-spec/power.json (hunk 15 above)
Comment (asadate) [reply part-spec/power.json]: yes. We moved this back

File: part-spec/power.json (hunk 14 above)
Comment (asadate) [reply part-spec/power.json]: changed

File: part-spec/power.json (hunk 16 above)
Comment (asadate) [reply part-spec/power.json:429]: Correct. Changed

File: part-spec/power.json (hunk 17 above)
Comment (asadate) [reply part-spec/power.json]: yes. Changed

File: part-spec/power.json (hunk 18 above)
Comment (asadate) [reply part-spec/power.json]: changed

File: part-spec/power.json (hunk 19 above)
Comment (asadate) [reply part-spec/power.json]: changed

File: part-spec/power.json (hunk 20 above)
Comment (asadate) [reply part-spec/power.json]: changed

File: part-spec/power.json (hunk 21 above)
Comment (asadate) [reply part-spec/power.json]: changed

**Comment by asadate** [comment]
Abandon PR and move changed to sensor3 branch (started from updated main)

**Pull request closed without merging**